config/config.go

```diff
@@ -1012,10 +1012,16 @@ func (cfg *ConsensusConfig) Precommit(round int32) time.Duration {
 	) * time.Nanosecond
 }
 
-// Commit returns the amount of time to wait for straggler votes after receiving +2/3 precommits
-// for a single block (ie. a commit).
-func (cfg *ConsensusConfig) Commit(t time.Time) time.Time {
-	return t.Add(cfg.TimeoutCommit)
+// Commit returns the amount of time to wait for straggler votes after receiving
+// +2/3 precommits for a single block (ie. a commit). This has been modified
+// from upstream tendermint to be dynamic, where we account for the amount of
+// time already taken in a round.
+func (cfg *ConsensusConfig) Commit(t time.Time, elapsed time.Duration) time.Time {
+	remaining := cfg.TargetRoundDuration - elapsed
```

Comment:
This looks like a reasonable, simple change to me. Would it make sense to:
 - test this in a live network or at least in testground
 - have informal review this as well
before we merge this?

Reply:
yes that's a good idea. We were trying to test this with testground this morning, but didn't have time to finish. 

We might not want to include this change in the incentivized testnet, but we should push and test in mocha as soon as we can get data back from testground.

One side thought: after playing around with this, I noticed that if we increase the timeouts too long, then validators begin to miss producing blocks, even on the e2e test. After discussing with @cmwaters, when the timeouts are long, we think there is a significantly higher probability validators are progressing through consensus at different rates, and thus potentially missing rounds entirely. This would actually explain what we are seeing in mocha quite well. Having more consistent round times, potentially via this mechansim, but likely via similar arbitary waiting mechanisms to more closely follow the configured timeouts instead of moving to the next step in consensus after reaching 2/3s, could be effective in helping us come to consensus on the first block when we expect.